### PR TITLE
feat: add search & filter, winget manifests, and publishing docs

### PR DESCRIPTION
**Pull request by lqdev**
## Summary

Adds search and filter functionality across all list buffers, winget package manifests, and publishing documentation.

### Search & Filter (Sprint 4)
- **Text search** (`/`) across all list buffers (episode list, What's New, podcast list)
- **Filter commands** via minibuffer:
  - `:filter-played` / `:filter-unplayed` — filter by play status
  - `:filter-date` — filter by date range (e.g., `today`, `7d`, `2025-01-01..2025-12-31`)
  - `:filter-podcast` — filter episodes by podcast name (with autocomplete)
  - `:clear-filters` / `F6` — remove all active filters
- **Filter indicator** in status bar showing active filter count
- **Help buffer** updated with Search & Filter section
- Duration filter deferred (Design Decision #13) — `extract_duration` is currently a stub

### Winget Publishing
- Add winget manifests for `lqdev.PodcastTUI` v1.5.0 (x64 + ARM64, zip portable)
- Create `docs/WINGET_PUBLISHING.md` with full publishing workflow documentation
- Update `BUILD_SYSTEM.md`, `README.md`, `AGENTS.md` with winget cross-references
- Fix README.md releases URL placeholder (`yourusername` → `lqdev`)

## Testing
- **166 unit tests** all passing
- New tests cover filter parsing, application, edge cases, and buffer integration
- No new clippy warnings introduced

## Files Changed (19 files, +2246 / -190)

| Area | Files |
|------|-------|
| Filter infrastructure | `src/ui/filters.rs` (new), `src/ui/mod.rs` |
| Buffer integration | `episode_list.rs`, `whats_new.rs`, `podcast_list.rs`, `help.rs` |
| App & keybindings | `src/ui/app.rs`, `src/ui/keybindings.rs` |
| Config | `src/config.rs`, `src/podcast/feed.rs` |
| Winget manifests | `manifests/l/lqdev/PodcastTUI/1.5.0/*` (3 files) |
| Documentation | `WINGET_PUBLISHING.md`, `SEARCH_AND_FILTER.md`, `KEYBINDINGS.md`, `BUILD_SYSTEM.md`, `README.md`, `AGENTS.md` |

### CODE REVIEW

File: src/ui/buffers/help.rs [hunk 1]
```diff
@@ -135,6 +135,13 @@ impl HelpBuffer {
             "  Auto-cleanup:            Runs on startup if cleanup_after_days".to_string(),
             "                           is set in config (default: 30 days)".to_string(),
             "".to_string(),
+            "SEARCH & FILTER (Episode lists):".to_string(),
+            "  / :search          Search episodes by title".to_string(),
```
Comment (lqdev) [inline src/ui/buffers/help.rs]: **Accepted** ✅ — Fixed in 9143f25.

Changed to `/ :search` (removed `C-s` reference). Only `/` is actually bound in `keybindings.rs`.

**Rationale for not adding `C-s`**: The project targets "simple, universal keybindings that work reliably across all terminal emulators" (per copilot-instructions.md). `C-s` sends XOFF (flow-control pause) in many terminals, which would make it unreliable across platforms. The `/` key is universal and already the standard search keybinding in vim-like TUI apps. If `C-s` is desired in the future, it should be added as a separate PR with terminal compatibility testing.

File: src/ui/filters.rs [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+            duration: None,
+            short_max_minutes: DEFAULT_SHORT_MAX_MINUTES,
+            long_min_minutes: DEFAULT_LONG_MIN_MINUTES,
+        }
+    }
+}
+
+impl EpisodeFilter {
+    /// Set configurable duration thresholds from user config.
+    ///
```
Comment (lqdev) [inline src/ui/filters.rs]: **Accepted** ✅ — Fixed in 9143f25.

Added validation: if `short_max >= long_min`, the method now silently returns without modifying thresholds (keeping defaults). This prevents illogical "medium" duration bands.

Also added 3 unit tests covering:
- Valid thresholds (accepted)
- Equal values (rejected, defaults kept)
- Reversed values (rejected, defaults kept)

Although duration filtering is deferred (Design Decision #13), this is good defensive coding for when it's eventually enabled. Total test count: 169 (166 → 169).
